spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java

```diff
@@ -0,0 +1,226 @@
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=toUpperCase|concatWithSelf",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(1000).getPayload())
+				.isEqualTo("HELLO FUNCTION:HELLO FUNCTION".getBytes(StandardCharsets.UTF_8));
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplier() {
+		ApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("1".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("2".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("3".getBytes(StandardCharsets.UTF_8));
+		//etc
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplierComposedWithSingleFunction() {
+		ApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number|concatWithSelf",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("11".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("22".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("33".getBytes(StandardCharsets.UTF_8));
+		//etc
+	}
+
+	@Test
+	public void testMessageSourceIsCreatedFromProvidedSupplierComposedWithMultipleFunctions() {
+		ApplicationContext context =
+				new SpringApplicationBuilder(
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number|concatWithSelf|mmultiplyByTwo",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("22".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("44".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("66".getBytes(StandardCharsets.UTF_8));
+		//etc
+	}
+
+	@EnableAutoConfiguration
+	@Import(ProvidedMessageSourceConfiguration.class)
+	public static class SupplierConfiguration {
+		AtomicInteger counter = new AtomicInteger();
```

Comment:
Need some code reformatting: its hard to read it in this non-wrapped with blank lines style.